client/src/js/services/grid/GridGrouping.js

```diff
@@ -98,9 +99,32 @@ function GridGroupingService(GridAggregators, uiGridGroupingConstants, Session,
     }
   }
 
-  function handleBatchSelection (){
+  function handleBatchSelection() {
     var gridApi = this.gridApi;
+    var gridRows = gridApi.selection.getSelectedGridRows();
+    var parents = {};
+
+    var hasSelections = gridApi.selection.getSelectedRows().length > 0;
+
+    gridRows.forEach(function (row) {
+      var parentRow = row.treeNode.parentRow;
+      if (parentRow.treeLevel === 0 && !parents[parentRow.uid]) {
```

Comment:
I know what this is because we've talked about it, but the code is hard to read as it is.  Could you move this check into a function?

```js
// this function identifies parent rows that we haven't seen yet
function isUnusedParentRow(row) {
  return row.treeLevel === 0 && !parents[row.uid];
}

gridRows.forEach(function (row) {
  var parentRow = row.treeNode.parentRow; 
  if (isUnusedParentRow(parentRow)) {
     // do something.
  }
});
```